src/components/CommentFooter/Buttons.js

```diff
@@ -0,0 +1,263 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { take, find } from 'lodash';
+import { injectIntl, FormattedNumber, FormattedMessage } from 'react-intl';
+import { Icon, Tooltip } from 'antd';
+import ReactionsModal from '../Reactions/ReactionsModal';
+import { getUpvotes, getDownvotes } from '../../helpers/voteHelpers';
+import { sortVotes } from '../../helpers/sortHelpers';
+import { calculatePayout } from '../../vendor/steemitHelpers';
+import USDDisplay from '../Utils/USDDisplay';
+import PayoutDetail from '../PayoutDetail';
+
+@injectIntl
+class Buttons extends React.Component {
+  static propTypes = {
+    intl: PropTypes.shape().isRequired,
+    user: PropTypes.shape().isRequired,
+    comment: PropTypes.shape().isRequired,
+    defaultVotePercent: PropTypes.number.isRequired,
+    editable: PropTypes.bool,
+    editing: PropTypes.bool,
+    replying: PropTypes.bool,
+    pendingVotes: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number,
+        percent: PropTypes.number,
+      }),
+    ),
+    onLikeClick: PropTypes.func,
+    onDislikeClick: PropTypes.func,
+    onReplyClick: PropTypes.func,
+    onEditClick: PropTypes.func,
+  };
+
+  static defaultProps = {
+    editable: false,
+    editing: false,
+    replying: false,
+    pendingVotes: [],
+    onLikeClick: () => {},
+    onDislikeClick: () => {},
+    onReplyClick: () => {},
+    onEditClick: () => {},
+  };
+
+  state = {
+    reactionsModalVisible: false,
+  };
+
+  handleShowReactions = () =>
+    this.setState({
+      reactionsModalVisible: true,
+    });
+
+  handleCloseReactions = () =>
+    this.setState({
+      reactionsModalVisible: false,
+    });
+
+  render() {
+    const {
+      intl,
+      user,
+      comment,
+      pendingVotes,
+      defaultVotePercent,
+      editable,
+      editing,
+      replying,
+    } = this.props;
+
+    const pendingVote = find(pendingVotes, { id: comment.id });
+    const pendingLike = pendingVote && (pendingVote.percent > 0 || pendingVote.vote === 'like');
+    const pendingDisLike =
+      pendingVote && (pendingVote.percent < 0 || pendingVote.vote === 'dislike');
+
+    const payout = calculatePayout(comment);
+
+    const upVotes = getUpvotes(comment.active_votes).sort(sortVotes);
+    const downVotes = getDownvotes(comment.active_votes)
+      .sort(sortVotes)
+      .reverse();
+
+    const totalPayout =
+      parseFloat(comment.pending_payout_value) +
+      parseFloat(comment.total_payout_value) +
+      parseFloat(comment.curator_payout_value);
+    const voteRshares = comment.active_votes.reduce((a, b) => a + parseFloat(b.rshares), 0);
```

Comment:
shouldnt this be `voterShares` ?

Reply:
They are actually called `rshares` so I think `voteRshares` is okay-ish name